src/app/cart/cart.service.ts

```diff
@@ -1,13 +1,56 @@
 import { Injectable } from '@angular/core';
-import { Product } from '../product/product.model';
 
 @Injectable()
 export class CartService {
-  products: Array<Product> = [];
+  products: Array<any> = [];
 
   constructor() { }
 
-  addToCart(product: Product) {
-    this.products.push(product);
+  addToCart(product: { id: number, name: string, price: number }) {
+    this.products.find(item => item.id === product.id)
+      ? this.updateQuantity(product.id)
+      : this.products.push({
+          id: product.id,
+          name: product.name,
+          price: product.price,
+          quantity: 1,
+        });
+  }
+
+  updateQuantity(id, way?) {
+    switch (way) {
+      case '-':
+        this.products = this.products.map(item => {
+          if (item.id === id && item.quantity > 1) {
+            item.quantity -= 1;
+          }
+          return item;
+        });
+        break;
+      case '+':
+      default:
+        this.products = this.products.map(item => {
+          if (item.id === id) {
+            item.quantity += 1;
```

Comment:
Why not `++` ?

Reply:
Because it's more obvious and I prefer it in this case. 
Tell this guy about ++:
https://github.com/douglascrockford

Reply:
Agree, it's a matter of taste